packages/@aws-cdk/aws-cloudfront-origins/lib/http-origin.ts

```diff
@@ -0,0 +1,24 @@
+import * as cloudfront from '@aws-cdk/aws-cloudfront';
+
+/**
+ * Properties for an Origin backed by any HTTP server.
+ *
+ * @experimental
+ */
+export interface HttpOriginProps extends cloudfront.HttpOriginOptions { }
+
+/**
+ * An Origin for an HTTP server.
+ *
+ * @experimental
+ */
+export class HttpOrigin extends cloudfront.HttpOrigin {
+
+  constructor(domainName: string, props: HttpOriginProps = {}) {
+    super({
+      domainName,
```

Comment:
How come `domainName` is not required in `cloudfront.HttpOriginOptions`? If it is, then you will likely need to split this type into two.

Reply:
This is what I called out in the PR description that I wanted feedback on.

Prior to this change, there was no `OriginOptions` and `OriginProps` had one required member: `domainName`. My thought for the higher-level Origins is that users shouldn't have to specify the domain name explicitly wherever possible -- it should be inferred from the construct. So `new S3Origin(bucket)` instead of `new S3Origin(bucket, {domainName: bucket.regionalBucketDomainName})`.

In the case, `cloudfront_origins.HttpOriginProps` could have just extended `cloudfront.HttpOriginProps`, but I wanted `domainName` here to be a positional parameter for consistency with the other origins that have one position parameter + the options.

So...

> How come domainName is not required in cloudfront.HttpOriginOptions?

Because I created the Options explicitly for the purpose of having an interface that matched the Props, minus domainName (due to all of the above).

I'm open to feedback on the above. I wanted to avoid having to create (for example) LoadBalancerV2Props that was the full set of OriginProps minus domainName.

Reply:
If domain name is required for any http origin why not just make it a positional argument of `cloudfront.HttpOrigin`?